src/js/components/DataTable/Header.js

```diff
@@ -195,35 +195,28 @@ const Header = forwardRef(
         (selected && enabled.filter((v) => selected.includes(v))) ||
         primaryValues;
 
-      const allSelected = groupBy?.select
-        ? groupBy.select[''] === 'all'
-        : enabledSelected.length === enabled.length;
-
-      if (allSelected) {
-        // if any are disabled and selected, leave those, otherwise clear
-        nextSelected = disabled
-          ? primaryValues.filter(
-              (v) => disabled.includes(v) && selected.includes(v),
-            )
-          : [];
-        nextGroupSelected[''] = 'none';
+      // if all enabled are already selected, remove them from selected;
+      // otherwise add them.
+      if (
+        selected.filter((s) => enabledSelected.includes(s)).length ===
```

Comment:
Just trying to sort out the logic here. Does `enabledSelected` already contain the filtered array of selected values? If so can we simply do `if (enabledSelect.length === enabled.length)`?

Reply:
done.